controllers/metal3.io/host_state_machine.go

```diff
@@ -107,11 +119,50 @@ func (hsm *hostStateMachine) updateHostStateFrom(initialState metal3v1alpha1.Pro
 			}
 		}
 	}
+
+	return true
 }
 
-func (hsm *hostStateMachine) ReconcileState(info *reconcileInfo) actionResult {
+func (hsm *hostStateMachine) checkDelayedHost(info *reconcileInfo) (res bool, actionRes actionResult) {
```

Comment:
Just from reading this function I have no idea what the first return value means.